File: GameEngine_IOS/GameEngine/Commons/Models/Light.m
#import "Light.h"

@implementation Light

/**
 * The constructor of the light entity
 *
 * @param aPosition
 *            Position where the light will exist
 * @param aColor
 *            The intensity of the light in the different components
 */
- (id)init : (Vector3f*) aPosition :  (Vector3f*) aColor
{
    self = [super init];
    if (self) {
        self -> _position = aPosition;
        self -> _color = aColor;
    }
    return self;
}



@end

File: GameEngine_IOS/GameEngine/Commons/Models/Light.swift
import Foundation

/**
 * Represents one light source in the scene
 */
open class Light : NSObject {
    
    /**
     * Position where the light will exist
     */
    var position : Vector3f;
    
    /**
     * The intensity of the light in the different components
     */
    var color : ColorRGB;
    
    /**
     * The constructor of the light entity
     *
     * @param aPosition
     *            Position where the light will exist
     * @param aColor
     *            The intensity of the light in the different components
     */
    public init(position : Vector3f, color : ColorRGB) {
        self.position = position
        self.color = color
    }
    
}
